plugins/mongodb/README.md

```diff
@@ -0,0 +1,202 @@
+
+```
+dataset({STRING first, STRING last, REAL salary}) findOne() := EMBED(mongodb : user(user), password(pwd), server(server), database(db),  collection(coll))
+    find_one({first: "John", last: "Doe"});
+ENDEMBED;
+findOne();
+
+dataset({STRING first, STRING last, REAL salary}) findMany() := EMBED(mongodb : user(user), password(pwd), server(server), database(db),  collection(coll))
+    find({first: "John", last: "Doe"});
+ENDEMBED;
+findMany();
+```
+
+There is support for providing a projection to the find function. If no projection is given then MongoDB returns all fields from the document, and the ECL engine must look through every field trying to find the ones declared in the return type by the ECL user.
+
+```
+dataset({STRING first, STRING last, REAL salary}) findMany(INTEGER include) := EMBED(mongodb : user(user), password(pwd), server(server), database(db),  collection(coll))
+    find({first: "John", last: "Doe"}, {first: $include, last: $include, salary: $include);
+ENDEMBED;
+findMany(1);
+```
+
+### Updating Documents
+
+The plugin can update one or many documents which is specified by the user. it returns a single Record with two fields holding the number of documents that were matched by the filter and another holding the number of documents that were modified by the operation. The update methods can take either two documents or a document and a pipeline. A pipeline follows the structure [{doc}, ... ].
+
+```
+dataset({Integer matched_count, Integer modified_count}) updateOne(REAL raise) := EMBED(mongodb : user(user), password(pwd), server(server), database(db),  collection(coll))
+    update_one({}, {$inc: {"salary": $raise}});
+ENDEMBED;
+updateOne(1000);
+
+dataset({Integer matched_count, Integer modified_count}) updateOne(REAL min, REAL raise) := EMBED(mongodb : user(user), password(pwd), server(server), database(db),  collection(coll))
+    update_many({"salary": {$lte: $min}}, {$inc: {"salary": $raise}});
+ENDEMBED;
+updateMany(100000, 3000);
+```
+
+The first example updates the first document that matches the find filter. In this example the find filter is empty, so all of the documents are returned. The raise argument gets inserted into the document as a real instead of a string.
+
+In the second example a filter is passed that will return all the documents that are below the min argument passed to the function. Then it will increase the salary of every document matching the filter by the raise argument passed into the function.
+
+
+### Deleting Documents
+
+The plugin can delete one or many documents which is specified by the user. It returns a document with a single field holding the number of deleted documents. The delete function takes a single document as a filter for the delete operation.
+
+
+```
+dataset({Integer deleted_count}) deleteOne(REAL min) := EMBED(mongodb : user(user), password(pwd), server(server), database(db),  collection(coll))
```

Comment:
Also a standardized RECORD definition, perhaps?

Reply:
This can be standard RECORD definition.